src/components/ClientOnly/ClientOnly.tsx

```diff
@@ -22,6 +22,8 @@ export const ClientOnly: ComponentWithChildren<ClientOnlyProps> = ({
 }) => {
   const [hasMounted, setHasMounted] = useState(false);
 
+  // FIXME: legitimate usecase - however ideally we wouldn't need this entire file
+  // eslint-disable-next-line no-restricted-syntax
```

Comment:
We would still need this component even if we switch over to RSC, server/client components, etc - because client components also get SSR'd and In the case where a component renders something that is totally dependant on client state - for example Theme button SVG icon - we would need to use ClientOnly here to render a "fallback" version during SSR and render the final version on client to avoid hydration errors. More on this - https://www.joshwcomeau.com/react/the-perils-of-rehydration/

Reply:
just move theme to a cookie in that example, but yes point taken

in an *ideal* world we wouldn't need this